dask/imperative.py

```diff
@@ -63,6 +64,12 @@ def to_task_dasks(expr):
     """
     if isinstance(expr, Value):
         return expr.key, expr._dasks
+    elif isinstance(expr, base.Base):
+        name = tokenize(str(expr), True)
+        keys = expr._keys()
+        dsk = expr._optimize(expr.dask, keys)
+        dsk[name] = (expr._finalize, expr, (concrete, keys))
```

Comment:
This is needed because `dask.array` needs to know the shape of the output to properly finalize it.